### PR TITLE
Update configuring-metricbeat.asciidoc

### PR DESCRIPTION
**Pull request by Leaf-Lin**
Adding ssl related configuration in the `modules.d/elasticsearch-xpack.yml` and the explanation because having a security-enabled cluster is a common scenario, yet, to allow metricbeat to send monitoring from a security-enabled cluster, one has to either use `ssl.verification_mode: "none"` (default is full) or to add other three ssl settings, namely, 
- ssl.certificate_authorities
- ssl.certificate
- ssl.key

The current document makes it very hard to find the related reference in metricbeat. One need to go to the elasticsearch module first,
https://www.elastic.co/guide/en/beats/metricbeat/7.8/metricbeat-module-elasticsearch.html
And from there, to check SSL settings which are available here:
https://www.elastic.co/guide/en/beats/metricbeat/7.8/configuration-ssl.html#_certificate_authorities

----
- Please check with developers on the correctness of the above when security is enabled on the host.
- Please feel free to rewording the paragraph.
- Please update the same docs to other versions too.

### CODE REVIEW

**Comment by Leaf-Lin** [comment]
Does the same thing apply in step 6? I haven't tested it, but from the look of it, the same steps appear to be relevant in step 6.

I also find the following document which is not very easy to find from all of the above references. But it is super useful to have a link to on this setup page. Can we have it somewhere?

https://www.elastic.co/guide/en/beats/metricbeat/7.8/securing-communication-elasticsearch.html

File: docs/reference/monitoring/configuring-metricbeat.asciidoc [hunk 1]
```diff
@@ -110,6 +115,8 @@ Alternatively, use the
 
 .. Add the `username` and `password` settings to the {es} module configuration
 file. 
+
+.. Add `ssl` related setting. By default ssl is disabled when the ssl section is missing. If Elastic security features are enabled, you must explicitly add {metricbeat-ref}/configuration-ssl.html[ssl section] to allow metricbeat to communicate to the host.
```
Comment (lcawl) [inline docs/reference/monitoring/configuring-metricbeat.asciidoc:119]:
```suggestion
.. Add `ssl` settings. By default, TLS/SSL is disabled when these settings are missing. If Elastic {security-features} are enabled, you must explicitly {metricbeat-ref}/configuration-ssl.html[configure SSL] to allow {metricbeat} to communicate to the host. 
```

**Comment by igoristic** [comment]
@Leaf-Lin @lcawl

I'm not sure I understand this part:
> If Elastic {security-features} are enabled, you must explicitly configure SSL to allow {metricbeat} to communicate to the host

Metricbeat should still be able to communicate with Elasticsearch even if `xpack.security.enabled: true` so long as you provide the right "Basic authentication" (username and password): https://www.elastic.co/guide/en/beats/metricbeat/current/elasticsearch-output.html#elasticsearch-output

**Comment by lcawl** [comment]
> It's only if ES also has SSL enabled ..."you must explicitly configure SSL to allow {metricbeat} to communicate to the host"

@igoristic Is it correct that it's only when TLS is enabled on the HTTP layer (https://www.elastic.co/guide/en/elasticsearch/reference/master/configuring-tls.html#tls-http) that these Metricbeat steps are necessary?

**Comment by tvernum** [comment]
Technically, the `must` here implies more than it should.
There are reasonable cases where the ES cluster may have TLS enabled, and the only change that is needed on metricbeat is to set the URL to be `https://`

I find some users get confused when docs say "you must" but in their config, they don't actually need to do it.

If the ES node's HTTP interface is using a certificate that is issued by a CA that is trusted by the operating system on which metricbeat is running, then all you need to do is enable SSL in some way (either by `ssl.enabled` or by setting a `https://` URL).

`ssl.certificate_authorities` would only be needed if the node's certificate is not trusted by the O/S cert store.
`ssl.certificate` and `ssl.key` would only be needed if the node is configured to use client certificates.

File: docs/reference/monitoring/configuring-metricbeat.asciidoc (hunk 1 above)
Comment (ycombinator) [reply docs/reference/monitoring/configuring-metricbeat.asciidoc:119]: Per @tvernum's comment, I think we should amend this section to say something like:

> If TLS is enabled on the HTTP layer of your Elasticsearch cluster, you must either use `https` as the URL scheme in the `hosts` setting or add the `ssl.enabled: true` setting. Depending on the TLS configuration of your Elasticsearch cluster, you may need to specify {metricbeat-ref}/configuration-ssl.html[additional `ssl.*`] settings.

**Review by ycombinator — APPROVED** [review]
LGTM.

@tvernum I've requested your review as well, as a double check. Thanks!

**Comment by ycombinator** [comment]
I tried re-targeting this PR to `master` instead of `7.8` but that resulted in a large number of conflicts. So I'm leaving this as-is. We'll just have to forward port the change to `master` after merging.

**Comment by lcawl** [comment]
> `ssl.certificate_authorities` would only be needed if...

If we want to go to this level of detail (as described by @tvernum), I think we could add that info as callouts on the snippet, same as #scope etc.

**Comment by tvernum** [comment]
> If we want to go to this level of detail 

To be clear, my intent in writing that detail was to make sure we had a shared understanding of the sorts of circumstances under which different configurations would be needed. I struggle to know whether that's helpful to put in the docs, or whether it's too much info (overwhelming to users).

My current feeling is that we need a page somewhere that helps people understand what all these SSL options do, and when you would need them. That's going to be hard to write well, but we can't keep avoiding it, because we're just pushing the problem to users, the community (forums etc) or support. If someone is configuring the stack with SSL enabled then they need guidance about how all of this works, and the docs are the place for that.

I don't think we want to repeat that level of info in every place that has some `ssl.*` configuration options (we'd just link to it), so my gut feel is not to included anything like that in this PR, and then open a separate issue where we can talk about _"How to explain the necessary details of SSL configuration helpfully to people who really just want their stack to run correctly"_.

**Review by tvernum — APPROVED** [review]
I think this is progress. The new info is useful & sufficiently accurate, and I don't think we should hold it up while we try and work out how to helpfully explain the next level of detail.

**Comment by lcawl** [comment]
@elasticmachine update branch

**Pull request merged**